From 5011f9b440c8c4144473d9a26171fc8ecf59d059 Mon Sep 17 00:00:00 2001
From: ravindu644 <droidcasts@protonmail.com>
Date: Sun, 22 Feb 2026 22:24:27 +0530
Subject: [PATCH] [PATCH 1/2] Droidspaces/LXC: fix: avoid kernel panic in
 xt_qtaguid by using rcu_read_lock around dev_get_stats

Signed-off-by: ravindu644 <droidcasts@protonmail.com>
---
 net/netfilter/xt_qtaguid.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/net/netfilter/xt_qtaguid.c b/net/netfilter/xt_qtaguid.c
index 8cb9c30bf..a1b2c3d45 100755
--- a/net/netfilter/xt_qtaguid.c
+++ b/net/netfilter/xt_qtaguid.c
@@ -748,12 +748,14 @@ static int iface_stat_fmt_proc_show(struct seq_file *m, void *v)
 
 	iface_entry = list_entry(v, struct iface_stat, list);
 
-	if (iface_entry->active) {
+	if (iface_entry->active && iface_entry->net_dev) {
+		rcu_read_lock();
 		stats = dev_get_stats(iface_entry->net_dev,
 				      &dev_stats);
+		rcu_read_unlock();
 	} else {
 		stats = &no_dev_stats;
 	}
 	/*
 	 * If the meaning of the data changes, then update the fmtX
 	 * string.
